client/wait.go

```diff
@@ -0,0 +1,248 @@
+			struct{}{},
+			&m,
+		)
+		if err != nil {
+			return true, fmt.Errorf("can't get status from daemon: %w", err)
+		}
+
+		t.report = m.Jobs
+
+		return t.evaluate()
+	}
+	updateAndEvaluate()
+
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
+ticker:
+	for range ticker.C {
+		done, err := updateAndEvaluate()
+		if err != nil {
+			return err
+		}
+		if done {
+			break ticker
+		}
+	}
+
+	return nil
+}
+
+func (t *wui) evaluate() (bool, error) {
+	v, ok := t.report[t.jobName]
+	if !ok {
+		return true, fmt.Errorf("job not in status: %v", t.jobName)
+	}
+
+	if v.Type == job.TypePush || v.Type == job.TypePull {
+		activeStatus, ok := v.JobSpecific.(*job.ActiveSideStatus)
+		if !ok || activeStatus == nil {
+			// TODO: ignore?
+			return true, nil //error.New("ActiveSideStatus is null")
+		}
+
+		//t.renderReplicationReport(activeStatus.Replication, t.getReplicationProgressHistory(k))
+		replicationDone, err := evaluateReplicationReport(activeStatus.Replication)
+		if err != nil {
+			return true, err
+		}
+
+		senderPruningDone, err := evaluatePruningReport(activeStatus.PruningSender)
+		if err != nil {
+			return true, fmt.Errorf("sender %w", err)
+		}
+		receiverPruningDone, err := evaluatePruningReport(activeStatus.PruningReceiver)
+		if err != nil {
+			return true, fmt.Errorf("receiver %w", err)
+		}
+
+		done := replicationDone && senderPruningDone && receiverPruningDone
+
+		if v.Type == job.TypePush {
+			snapperDone, err := evaluateSnapperReport(activeStatus.Snapshotting)
+			if err != nil {
+				return true, err
+			}
+
+			done = done && snapperDone
+		}
+
+		return done, nil
+	} else if v.Type == job.TypeSnap {
+		snapStatus, ok := v.JobSpecific.(*job.SnapJobStatus)
+		if !ok || snapStatus == nil {
+			// TODO: ignore?
+			return true, nil //error.Error("SnapJobStatus is null")
+		}
+
+		pruningDone, err := evaluatePruningReport(snapStatus.Pruning)
+		if err != nil {
+			return true, err
+		}
+
+		snapperDone, err := evaluateSnapperReport(snapStatus.Snapshotting)
+		if err != nil {
+			return true, err
+		}
+
+		return pruningDone && snapperDone, nil
+	} else if v.Type == job.TypeSource || v.Type == job.TypeSink {
+		snapStatus, ok := v.JobSpecific.(*job.PassiveStatus)
+		if !ok || snapStatus == nil {
+			// TODO: ignore?
+			return true, nil //error.Error("SnapJobStatus is null")
+		}
+
+		snapperDone, err := evaluateSnapperReport(snapStatus.Snapper)
+		if err != nil {
+			return true, err
+		}
+
+		return snapperDone, nil
+
+	} else {
+		return true, fmt.Errorf("unknown job type")
+	}
+}
+
+func evaluatePruningReport(r *pruner.Report) (bool, error) {
+	if r == nil {
+		// ignore
+		return true, nil
+	}
+
+	if r.Error != "" {
+		return true, fmt.Errorf("pruning error: %v", r.Error)
+	}
+
+	prunerState, err := pruner.StateString(r.State)
+	if err != nil {
+		return true, fmt.Errorf("parsing pruning state %v: %w", r.State, err)
+	}
+
+	switch prunerState {
```

Comment:
we should have a `prunerState.IsTerminal()` getter, use it both here and in zrepl status